src/test/scala/com/microsoft/hyperspace/index/IcebergIntegrationTest.scala

```diff
@@ -0,0 +1,190 @@
+          if (version.isDefined) {
+            val iceDf =
+              spark.read.format("iceberg").option("snapshot-id", version.get).load(dataPath)
+            iceDf.filter(iceDf("clicks") <= 2000).select(iceDf("query"))
+          } else {
+            val iceDf = spark.read.format("iceberg").load(dataPath)
+            iceDf.filter(iceDf("clicks") <= 2000).select(iceDf("query"))
+          }
+        }
+
+        assert(isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex"))
+
+        val currentIceSnapshot = iceTable.currentSnapshot().snapshotId()
+
+        // Create a new version by adding entries.
+        val dfFromSample2 = sampleData2.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        dfFromSample2.write.format("iceberg").mode("append").save(dataPath)
+
+        // The index should not be applied for the updated version.
+        assert(!isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex"))
+
+        // The index should be applied for the version at index creation.
+        assert(isIndexUsed(
+          query(Some(currentIceSnapshot)).queryExecution.optimizedPlan, "iceIndex"))
+
+        hyperspace.refreshIndex("iceIndex")
+
+        // The index should be applied for the updated version.
+        assert(isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex/v__=1"))
+
+        // The index should not be applied for the version at index creation.
+        assert(!isIndexUsed(query(Some(currentIceSnapshot)).queryExecution.optimizedPlan,
+          "iceIndex"))
+      }
+    }
+  }
+
+  test("Verify Hybrid Scan on Iceberg table.") {
+    withTempPathAsString { dataPath =>
+      import spark.implicits._
+      val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+
+      createIcebergTable(dataPath, dfFromSample)
+      dfFromSample.write.format("iceberg").mode("overwrite").save(dataPath)
+
+      val iceDf = spark.read.format("iceberg").load(dataPath)
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+        hyperspace.createIndex(iceDf, IndexConfig("iceIndex", Seq("clicks"), Seq("Query")))
+      }
+
+      withIndex("iceIndex") {
+        def query(version: Option[Long] = None): DataFrame = {
+          if (version.isDefined) {
+            val iceDf =
+              spark.read.format("iceberg").option("snapshot-id", version.get).load(dataPath)
+            iceDf.filter(iceDf("clicks") <= 2000).select(iceDf("query"))
+          } else {
+            val iceDf = spark.read.format("iceberg").load(dataPath)
+            iceDf.filter(iceDf("clicks") <= 2000).select(iceDf("query"))
+          }
+        }
+
+        assert(isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex", false))
+
+        // Create a new version by adding entries.
+        val dfFromSample2 = sampleData2.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        dfFromSample2.write.format("iceberg").mode("append").save(dataPath)
+
```

Comment:
Could you add 
`assert(!isIndexUsed(query().queryExecution.optimizedPlan, "iceIndex", true))` here? to test if the index is not applied for modified dataset.

Reply:
👍